go-controller/pkg/ovn/base_network_controller_policy.go

```diff
@@ -1410,11 +1418,83 @@ func (bnc *BaseNetworkController) handlePeerNamespaceSelectorAdd(np *networkPoli
 			errors = append(errors, err)
 		}
 	}
+	// check if UDN label is present on netpol namespace.
+	ns, err := bnc.watchFactory.GetNamespace(np.namespace)
+	if err != nil {
+		errors = append(errors, err)
+	}
+	if _, exists := ns.Labels[types.RequiredUDNNamespaceLabel]; !exists {
+		// Netpol is created on the default network namespace, so address set
+		// for the namespace must be present at the time of namespace creation.
+		// so return it now without subscribing to its pod events.
+		klog.V(5).Infof("Skipping pod event handling for network policy %s on namespace peer %s because it's managed by default primary network",
+			np.name, ns.Name)
+		return utilerrors.Join(errors...)
+	}
+	syncFunc := func(objs []interface{}) error {
+		// ignore returned error, since any pod that wasn't properly handled will be retried individually.
+		_ = bnc.handlePeerNamespacePodAddUpdate(np, gp, objs...)
+		return nil
+	}
+	retryFramework := bnc.newNetpolRetryFramework(
+		factory.AddressSetPodSelectorType,
+		syncFunc,
+		&NetworkPolicyExtraParameters{gp: gp, np: np},
+		np.cancelableContext.Done(),
+	)
+	for _, namespace := range namespaces {
+		if _, ok := np.nsPodHandlers[namespace]; !ok {
+			podHandler, err := retryFramework.WatchResourceFiltered(namespace, nil)
+			if err != nil {
+				errors = append(errors, err)
+			} else {
+				np.nsPodHandlers[namespace] = podHandler
+			}
+		}
+	}
 	return utilerrors.Join(errors...)
+}
 
+// handles pod add and update events (or) syncing existing pods on the namespace, it updates ACL(s) with namespace address
+// set if it needed and guarantees idempotent behavior.
+func (bnc *BaseNetworkController) handlePeerNamespacePodAddUpdate(np *networkPolicy, gp *gressPolicy, objs ...interface{}) error {
+	if config.Metrics.EnableScaleMetrics {
+		start := time.Now()
+		defer func() {
+			duration := time.Since(start)
+			metrics.RecordPodSelectorAddrSetPodEvent("add", duration)
+		}()
+	}
+	np.RLock()
+	if np.deleted {
+		np.RUnlock()
+		return nil
+	}
+	updated := false
+	var errors []error
+	for _, obj := range objs {
+		pod := obj.(*corev1.Pod)
+		// addNamespaceAddressSet is safe for concurrent use, doesn't require additional synchronization
+		nsUpdated, err := gp.addNamespaceAddressSet(pod.Namespace, bnc.addressSetFactory)
```

Comment:
The namespace address set might not exist yet. Is there any state on the pod that we can access (cheaply) that we can use if the address set might exist already? To avoid spurious errors and retries.

Reply:
That's not exactly what I suggested and incurs in locking and might actually be more expensive. If there is no `state on the pod that we can access (cheaply) `, then perhaps it is just better to avoid retrying if `addNamespaceAddressSet` returns a libovsdb not found error, as I suggested o the previous comment as well?